external/test-runtime/optional.csproj

```diff
@@ -1,6 +1,10 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <TargetFramework>net45</TargetFramework>
+    <!-- $(RestoreSources) is set in the root Directory.Build.props and will override any sources specified
```

Comment:
I thought the nuget.config and the RestoreSources get aggregated is that not the case?

Reply:
Does not seem like it. I just tried to repro it locally:

csproj:
```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp2.1</TargetFramework>
    <RestoreSources>$(RestoreSources);https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json</RestoreSources>
  </PropertyGroup>

<Target Name="TestMessage" AfterTargets="Build" >
    <Message Text="$(RestoreSources)" Importance="high"/>
  </Target>
</Project>
```

nuget:
```
<configuration>
    <packageSources>
        <add key="nuget.org" value="https://api.nuget.org/v3/index.json" protocolVersion="3" />
        <add key="Contoso" value="https://contoso.com/packages/" />
    </packageSources>
</configuration>
```

Output: `;https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json`

Reply:
Related: https://github.com/dotnet/docs/blob/master/docs/core/tools/dotnet-restore.md

> -s|--source <SOURCE>
Specifies a NuGet package source to use during the restore operation. This setting overrides all of the sources specified in the NuGet.config files. Multiple sources can be provided by specifying this option multiple times.

Reply:
Ah found it: https://github.com/NuGet/Home/wiki/%5BSpec%5D-NuGet-settings-in-MSBuild#order-of-precedence

> If a project provides a value for a property it will be used instead of NuGet.Config settings. If no value is provided then NuGet.Config will be read.

Reply:
> If a project provides a value for a property it will be used instead of NuGet.Config settings. If no value is provided then NuGet.Config will be read.

One place where this doesn't appear to be the case is `$(RestorePackagesPath)`.  When specifying `--configfile`, it appears the project property will be ignored.

Reply:
Maybe I'm missing something but this is exactly the behavior described in the link:

Settings will be determined using the following order.
1. Properties passed on the command line
2. MSBuild properties
3. NuGet.Config settings

Reply:
Not `--configFile` but `--source` overwrites `RestoresSource`. See the following example:

```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp2.1</TargetFramework>
    <RestoreSources>$(RestoreSources);https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json</RestoreSources>
  </PropertyGroup>

<Target Name="TestMessage" AfterTargets="Build" >
    <Message Text="$(RestoreSources)" Importance="high"/>
  </Target>
</Project>
```

> dotnet build --source C:\git\
```cmd
C:\git\
```

> dotnet build
```cmd
;https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json
```